### DISCUSSION TITLE
Corruption: block checksum mismatch error

### DISCUSSION THREAD

**Discussion by Yscream** [apache/kvrocks#2622] [Q&A] [answered]
Hi! i have a problem with kvrocks
`I20241024 11:49:21.631152    53 compaction_checker.cc:37] [compaction checker] Compact the column family: pubsub finished, result: Corruption: block checksum mismatch: stored = 3301255889, computed = 1963786118, type = 4  in /var/lib/kvrocks/data/db/001683.sst offset 63901683 size 25113525`
please tell me how to fix it

**Comment by git-hulk** [comment, marked as answer]
@Yscream which version you're using? Kvrocks will try to recover from this error since 2.6.0, please refer to: https://github.com/apache/kvrocks/pull/1667

**Reply by ltagliamonte-dd** [reply]
@Yscream i was able to fix it using ldb:
```
/opt/rocksdb/ldb repair --db=/data/kvrocks/db/
```

you can compile ldb following this instructions:
https://github.com/facebook/rocksdb/issues/13053#issuecomment-2403492498
